pandas/tests/io/excel/test_readers.py

```diff
@@ -148,6 +149,53 @@ def parser(self, *args, **kwargs):
             expected = expected_defaults[read_ext[1:]]
         assert result == expected
 
+    def test_engine_kwargs_valid(self, read_ext, engine):
+        # GH#52214
+        expected_defaults = {
+            "xlsx": {"keep_vba": True},
+            "xlsm": {"keep_vba": True},
+            "xlsb": {"debug": True},
+            "xls": {"ignore_workbook_corruption": False},
+            "ods": {},
+        }
+
+        if engine is not None:
+            df = pd.read_excel(
+                "test1" + read_ext,
+                sheet_name="Sheet1",
+                index_col=0,
+                **expected_defaults[read_ext[1:]],
+            )
+            assert not df.empty
+
+    def test_engine_kwargs_invalid(self, read_ext, engine):
+        # GH#52214
+        expected_defaults = {
+            "xlsx": {"foo": "abcd"},
+            "xlsm": {"foo": 123},
+            "xlsb": {"foo": "True"},
+            "xls": {"foo": True},
+            "ods": {},
+        }
+
+        # odf.opendocument does not take any additional args
+        if read_ext[1:] != "ods":
```

Comment:
Why are we excluding ods here?

Reply:
@rhshadrach The `load` method used to read in .ods in for _odfreader.py takes 1 and only 1 argument

Reply:
I see, we should still be passing engine_kwargs through. It may not in the future, and if a user tries to pass engine_kwargs today with that reader, their code should fail.

Reply:
Understood! Will fix this!